### PR TITLE
Use enums instead of symbols for `Spec`-related types

### PR DESCRIPTION
**Pull request by HertzDevil**
Those types (`Spec::RootContext` and the formatters) are not part of the public API, so there are no breaking changes.

### CODE REVIEW

File: src/spec/dsl.cr [hunk 1]
```diff
@@ -2,27 +2,45 @@ require "colorize"
...
-  def self.color(str, status)
+  def self.color(str, status : Status)
+    if use_colors?
+      str.colorize(STATUS_COLORS[status])
+    else
+      str
+    end
+  end
+
+  def self.color(str, kind : InfoKind)
```
Comment (oprypin) [inline src/spec/dsl.cr:41]: Probably should be nodoc too?
```suggestion
  # :nodoc:
  def self.color(str, kind : InfoKind)
```

File: src/spec/junit_formatter.cr [hunk 2]
```diff
@@ -89,10 +89,10 @@ module Spec
...
     private def inner_content_tag(kind)
       case kind
-      when :error   then "error"
-      when :fail    then "failure"
-      when :pending then "skipped"
-      else               nil
+      when .error?   then "error"
+      when .fail?    then "failure"
+      when .pending? then "skipped"
+      else                nil
```
Comment (oprypin) [inline src/spec/junit_formatter.cr:95]: Something to consider
```suggestion
      in .error?   then "error"
      in .fail?    then "failure"
      in .pending? then "skipped"
      in .success? then nil
```

File: src/spec/junit_formatter.cr [hunk 3]
```diff
@@ -104,7 +104,7 @@ module Spec
         HTML.escape(message, io)
         io << '"'
       end
-      if tag == :error
```
Comment (oprypin) [inline src/spec/junit_formatter.cr:107]: Wait what? Was this a bug?

Comment (HertzDevil) [reply src/spec/junit_formatter.cr:107]: Spec:

```crystal
require "spec"

it "errors" { raise "foo" }
```

Before:

```xml
<?xml version="1.0"?>
<testsuite tests="1" skipped="0" errors="1" failures="0" time="0.000237095" timestamp="2021-12-16T11:42:12Z" hostname="debian">
  <testcase file="/home/quinton/crystal/crystal/bug.cr" classname="bug" name="errors" time="7.5482000000000007e-05">
    <error message="foo">bug.cr:75:15 in &#39;-&gt;&#39;
src/primitives.cr:266:3 in &#39;internal_run&#39;
src/spec/example.cr:33:16 in &#39;run&#39;
src/spec/context.cr:18:23 in &#39;internal_run&#39;
src/spec/context.cr:156:7 in &#39;run&#39;
src/spec/dsl.cr:220:7 in &#39;-&gt;&#39;
src/primitives.cr:266:3 in &#39;run&#39;
src/crystal/main.cr:50:14 in &#39;exit&#39;
src/crystal/main.cr:45:5 in &#39;main&#39;
src/crystal/main.cr:124:3 in &#39;main&#39;
/lib/x86_64-linux-gnu/libc.so.6 in &#39;__libc_start_main&#39;
/home/quinton/.cache/crystal/crystal-run-spec.tmp in &#39;_start&#39;
???</error>
  </testcase>
</testsuite>
```

After:

```xml
<?xml version="1.0"?>
<testsuite tests="1" skipped="0" errors="1" failures="0" time="0.000288758" timestamp="2021-12-16T11:42:31Z" hostname="debian">
  <testcase file="/home/quinton/crystal/crystal/bug.cr" classname="bug" name="errors" time="7.3282e-5">
    <error message="foo" type="Exception">bug.cr:75:15 in &#39;-&gt;&#39;
src/primitives.cr:266:3 in &#39;internal_run&#39;
src/spec/example.cr:33:16 in &#39;run&#39;
src/spec/context.cr:18:23 in &#39;internal_run&#39;
src/spec/context.cr:156:7 in &#39;run&#39;
src/spec/dsl.cr:220:7 in &#39;-&gt;&#39;
src/primitives.cr:266:3 in &#39;run&#39;
src/crystal/main.cr:50:14 in &#39;exit&#39;
src/crystal/main.cr:45:5 in &#39;main&#39;
src/crystal/main.cr:124:3 in &#39;main&#39;
/lib/x86_64-linux-gnu/libc.so.6 in &#39;__libc_start_main&#39;
/home/quinton/.cache/crystal/crystal-run-spec.tmp in &#39;_start&#39;
???</error>
  </testcase>
</testsuite>
```

I have updated a spec for this.

**Review by oprypin — APPROVED** [review]

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**